test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -205,8 +207,9 @@ void doBlockchainTestNoLog(json_spirit::mValue& _v, bool _fillin)
 				tests[newtestname] = jObj;
 			}
 
-			//delete source test from the json
-			_v.get_obj().erase(_v.get_obj().find(testname));
```

Comment:
Do you remove the current item? If yes, you should keep the iterator somewhere and use the common pattern: https://stackoverflow.com/questions/8234779/how-to-remove-from-a-map-while-iterating-it.